capstone-app/src/app/modules/maps/map/map.component.html

```diff
@@ -1,63 +1,115 @@
+      <button
+      class="dropdown"
+      mat-icon-button
+      (click)="toggleOptions()"
+    >
+      <mat-icon>{{!showOptions ? 'expand_more' : 'expand_less'}}</mat-icon>
+    </button>
+      <mat-form-field class="search-bar">
         <mat-label>Search for a city</mat-label>
-        <input matInput type="text" #searchBar>
-        <button mat-button
+        <input matInput type="text" #searchBar />
+        <button
           *ngIf="searchBar.value"
-          matSuffix mat-icon-button aria-label="Clear"
-          (click)="searchBar.value=''">
-            <mat-icon>close</mat-icon>
+          matSuffix
+          mat-icon-button
+          aria-label="Clear"
+          (click)="searchBar.value = ''"
+        >
+          <mat-icon>close</mat-icon>
         </button>
-    </mat-form-field>
-    <button class="search-icon" mat-icon-button color="primary" (click)="locationSearch()" >
+      </mat-form-field>
+      <button
+      class="search-icon"
+      mat-icon-button
+      (click)="locationSearch()"
+    >
       <mat-icon>search</mat-icon>
     </button>
-  </div>
-
-  <mat-form-field appearance="fill">
-    <mat-label>Map Theme</mat-label>
-    <mat-select id="map-theme" (selectionChange)="setMapStyle($event.value)">
-      <mat-option *ngFor="let map of mapTypes" [value]="map">
-        {{map}}
-      </mat-option>
-    </mat-select>
-  </mat-form-field>
+    </mat-list-item> 
+    <div [ngClass]="showOptions ? 'show' : 'hidden'">
+    <mat-divider></mat-divider>
+    <mat-list-item role="listitem">
+      <ng-template matMenuContent></ng-template>
+      <div class="flex-menu">
+      <span [matMenuTriggerFor]="menu">
+        <mat-icon>map</mat-icon>
+        Map Theme
+      </span>
+    </div>
+      <mat-menu #menu="matMenu">
+        <ng-container *ngFor="let map of mapTypes">
+          <button mat-menu-item (click)="setMapStyle(map)">
+            <mat-icon>explore</mat-icon>
+            <span> {{ map }}</span>
+          </button>
+        </ng-container>
+      </mat-menu>
+    </mat-list-item>
 
-  <mat-form-field appearance="fill">
-    <mat-label class="mat-select">Marker Types</mat-label>
-    <mat-select id="type-select" (selectionChange)="changeType($event.value)">
-      <mat-option *ngFor="let option of optionsTypes" [value]="option">
-        {{option}}
-      </mat-option>
-    </mat-select>
-  </mat-form-field>
-
-  <button class="button-delete" mat-raised-button (click)="deleteAllCustomMark()">Clear</button>
-</span>
+    <mat-list-item role="listitem">
+      <ng-template matMenuContent></ng-template>
+      <div class="flex-menu">
+      <span [matMenuTriggerFor]="menu">
+        <mat-icon>menu</mat-icon>
+        Marker Types
+      </span>
+    </div>
+      <mat-menu #menu="matMenu">
+        <ng-container *ngFor="let option of optionsTypes">
+          <button mat-menu-item (click)="changeType(option[0])">
+            <mat-icon>{{option[1]}}</mat-icon>
```

Comment:
add spaces between option[1] and curly braces and is there a reason why the icon is set to option[1] but the span is for option[0] wouldn't be better for both icon and span to be the option item.

Reply:
The string in between the mat-icon represents the icon code for the icon. For example, a globe is explore. I handpicked icons for each type and put them in a map next to the options